src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToControlGeneratedAssemblyInfo.cs

```diff
@@ -464,6 +464,68 @@ public void It_doesnt_includes_requires_preview_features()
             Assert.False(contains);
         }
 
+        [Theory]
+        [InlineData(true, true, "net6.0", false)]
+        [InlineData(true, false, "net6.0", false)]
+        [InlineData(true, null, "net6.0", false)]
+        [InlineData(false, false, "net6.0", false)]
+        [InlineData(false, null, "net6.0", false)]
+        [InlineData(true, true, "net7.0", true)]
+        [InlineData(true, false, "net7.0", false)]
+        [InlineData(true, null, "net7.0", true)]
+        [InlineData(false, false, "net7.0", false)]
+        [InlineData(false, null, "net7.0", false)]
+        [InlineData(true, true, ToolsetInfo.CurrentTargetFramework, true)]
+        [InlineData(true, false, ToolsetInfo.CurrentTargetFramework, false)]
+        [InlineData(true, null, ToolsetInfo.CurrentTargetFramework, true)]
+        [InlineData(false, false, ToolsetInfo.CurrentTargetFramework, false)]
+        [InlineData(false, null, ToolsetInfo.CurrentTargetFramework, false)]
+        public void TestDisableRuntimeMarshalling(bool disableRuntimeMarshalling, bool? generateDisableRuntimeMarshallingAttribute, string targetFramework, bool shouldHaveAttribute)
+        {
+            var testProject = new TestProject()
+            {
+                Name = "HelloWorld",
+                TargetFrameworks = targetFramework,
+                IsExe = true,
+            };
+            testProject.AdditionalProperties["DisableRuntimeMarshalling"] = disableRuntimeMarshalling.ToString();
+            if (generateDisableRuntimeMarshallingAttribute.HasValue)
+            {
+                testProject.AdditionalProperties["GenerateDisableRuntimeMarshallingAttribute"] = generateDisableRuntimeMarshallingAttribute.Value.ToString();
+            }
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
```

Comment:
This should make each instance of the theory use a different test execution folder, and fix the test failures in CI:

```suggestion
            var testAsset = _testAssetsManager.CreateTestProject(testProject, identifier: disableRuntimeMarshalling.ToString() + "_" + generateDisableRuntimeMarshallingAttribute + "_" + targetFramework);
```